lib/si-id/src/lib.rs

```diff
@@ -1,242 +1,75 @@
-            fn to_sql(
-                &self,
-                ty: &postgres_types::Type,
-                out: &mut postgres_types::private::BytesMut,
-            ) -> Result<postgres_types::IsNull, Box<dyn std::error::Error + Sync + Send>>
-            where
-                Self: Sized,
-            {
-                postgres_types::ToSql::to_sql(&self.0.to_string(), ty, out)
-            }
-
-            fn accepts(ty: &postgres_types::Type) -> bool
-            where
-                Self: Sized,
-            {
-                ty == &postgres_types::Type::BPCHAR
-                    || ty.kind() == &postgres_types::Kind::Domain(postgres_types::Type::BPCHAR)
-            }
-
-            fn to_sql_checked(
-                &self,
-                ty: &postgres_types::Type,
-                out: &mut postgres_types::private::BytesMut,
-            ) -> Result<postgres_types::IsNull, Box<dyn std::error::Error + Sync + Send>> {
-                postgres_types::ToSql::to_sql(&self.0.to_string(), ty, out)
-            }
-        }
-    };
-}
-
-impl From<PropId> for PropertyEditorPropId {
-    fn from(prop_id: PropId) -> Self {
-        Self::from(::ulid::Ulid::from(prop_id))
-    }
-}
-
-impl From<PropertyEditorPropId> for PropId {
-    fn from(property_editor_prop_id: PropertyEditorPropId) -> Self {
-        Self::from(::ulid::Ulid::from(property_editor_prop_id))
-    }
-}
-
-impl From<AttributeValueId> for PropertyEditorValueId {
-    fn from(id: AttributeValueId) -> Self {
-        Self::from(::ulid::Ulid::from(id))
-    }
-}
-
-impl From<PropertyEditorValueId> for AttributeValueId {
-    fn from(id: PropertyEditorValueId) -> Self {
-        Self::from(::ulid::Ulid::from(id))
-    }
-}
-
-id!(ActionId);
+// Please keep these alphabetically sorted!
 id!(ActionPrototypeId);
 id!(AttributePrototypeArgumentId);
 id!(AttributePrototypeId);
-id!(AttributeValueId);
 id!(AuthenticationPrototypeId);
-id!(CachedModuleId);
-id!(ChangeSetId);
-id!(ComponentId);
 id!(DeprecatedVectorClockId);
 id!(EventSessionId);
 id!(FuncArgumentId);
 id!(FuncExecutionPk);
-id!(FuncId);
-id!(FuncRunId);
 id!(GeometryId);
 id!(HistoryEventPk);
 id!(InputSocketId);
-id!(KeyPairPk);
 id!(ManagementPrototypeId);
 id!(ModuleId);
 id!(NaxumApiTypesRequestId);
 id!(OutputSocketId);
 id!(PropId);
 id!(PropertyEditorPropId);
 id!(PropertyEditorValueId);
-id!(SchemaId);
-id!(SchemaVariantId);
 id!(SecretId);
 id!(StaticArgumentValueId);
-id!(UserPk);
 id!(ValidationOutputId);
 id!(ViewId);
-id!(WorkspaceId);
-id!(WorkspacePk);
 id!(WorkspaceSnapshotNodeId);
+
+// Please keep these alphabetically sorted!
+id_with_pg_types!(ActionId);
+id_with_pg_types!(CachedModuleId);
+id_with_pg_types!(ChangeSetId);
+id_with_pg_types!(ComponentId);
+id_with_pg_types!(FuncId);
+id_with_pg_types!(FuncRunId);
+id_with_pg_types!(SchemaId);
+id_with_pg_types!(UserPk);
+
+// Please keep these alphabetically sorted!
+id_with_none!(SchemaVariantId);
+
+// Please keep these alphabetically sorted!
```

Comment:
This comment appears multiple times and I think there's a world where we could read from an enum with the `#[remain::sorted]` macro to handle sorting here within another macro, but for now, this should suffice.